app/src/main/java/com/hillnerds/soundsparrow/ChannelSoundGeneration.java

```diff
@@ -0,0 +1,177 @@
+
+/**
+ * Created by aga on 06/02/17.
+ */
+
+public class ChannelSoundGeneration {
+
+    int scale;
+    int[] steps;
+
+    //a hard coded list of all possible instruments
+    private static Instrument[] instrumentList = new Instrument[] {
+            new Instrument("piano", 1, 40, 60),
+            new Instrument("guitar", 26, 45, 70),
+            new Instrument("trombone", 58, 55, 70),
+            new Instrument("trumpet", 57, 64, 80),
+            new Instrument("violin", 41, 50, 65),
+            new Instrument("saxophone", 66, 45, 60),
+            new Instrument("flute", 74, 73, 90)
+    };
+
+    /**
+     * This function chooses a rondom instrument form a hardcoded instrumentList.
+     * @param random - a random generator Object
+     * @return - return an Instrument object from instrumentList
+     */
+    public static Instrument generateRandomInstrument(Random random){
+        int instrumentNumber = random.nextInt((6 - 0) + 1) + 0;
+        return instrumentList[instrumentNumber];
+    }
+
+    /**
+     * Generate a MIDI file for a single Channel.
+     * @param c - A single channel object
+     * @return - An array of MidiSequence Objects
+     */
+    public static ArrayList<MidiSequence> generateMidiChannel(Channel c) {
+        ArrayList<MidiSequence> midiSequenceArray = new ArrayList<>();
+
+        int timestampCounter = 0;
+
+        StartingSequence instrument_change = new StartingSequence(192 + c.number, c.instrument, timestampCounter);
+        midiSequenceArray.add(instrument_change);
+
+        /**
+         * TODO: implement the way of setting the length of the track differently (ie. a number of
+         * bars so that tunes are synchronised)
+         */
+        for (int i = 0; i < 16; i = i+2) {
+            int startingCode = generateStartingCode(c.number);
+            int pitch = generatePitch(c);
+            int velocity = generateVelocity(c.randomGenerator);
+
+            int duration = generateNoteDuration(c.randomGenerator);
+
+            //create a Note on sequence (start of the note)
+            Note note_on = new Note(startingCode, pitch, velocity, timestampCounter);
+            timestampCounter = timestampCounter + duration;
+
+            //create a Note off sequence (same as Note on but with velocity at 0)
+            Note note_off = new Note(startingCode, pitch, 0, timestampCounter);
+
+            midiSequenceArray.add(note_on);
+            midiSequenceArray.add(note_off);
+        }
+        return midiSequenceArray;
+    }
+
+    /**
+     * Genrates a starting code for a note. Of the form XXXX CCCC where XXXX is the code
+     * of the MidiSequnce (ie. 1001 Note on) and CCCC is the channel number (from 0 to 15)
+     * @param channelNumber - the number of tha channel for which the starting seqence will be generated
+     * @return - a total starting code value
+     */
+    public static int generateStartingCode(int channelNumber){
+        return 144 + channelNumber;
```

Comment:
What exactly is 144? It's good that this is in its own documented function though.